### PR TITLE
Generate User and Loan Models

### LINKED ISSUE #2: Create User model
Acceptance criteria:
- First name / Last name
- email (unique)

### PR DESCRIPTION
**Pull request by ernest-phillips**
This closes #2 and #3 

This PR adds the User and Loan models for our application with the initial values for the acceptance critieria.

### CODE REVIEW

File: app/controllers/loans_controller.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class LoansController < ApplicationController
+  def new
```
Comment (gabrielgaraujo) [inline app/controllers/loans_controller.rb:2]: Please, let's change `new` for create

The `new` action is used to [serve forms views](https://guides.rubyonrails.org/action_controller_overview.html#methods-and-actions)

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class UsersController < ApplicationController
+  def new
```
Comment (gabrielgaraujo) [inline app/controllers/users_controller.rb:2]: Same as https://github.com/ernest-phillips/debt-tracker/pull/7/files?w=1#r947172831

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class User < ApplicationRecord
+
```
Comment (gabrielgaraujo) [inline app/models/user.rb:2]: We can have a `email` validation

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'users/new'
```
Comment (gabrielgaraujo) [inline config/routes.rb:2]: These are not RESTful you can read more about RESTful APIs [here](https://www.restapitutorial.com/lessons/httpmethods.html#:~:text=The%20primary%20or%20most-commonly,but%20are%20utilized%20less%20frequently.)
Let's also use the [`resources`](https://guides.rubyonrails.org/routing.html#configuring-the-rails-router) to generate our routes

File: db/migrate/20220816183243_create_users.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+class CreateUsers < ActiveRecord::Migration[7.0]
+  def change
+    create_table :users do |t|
+      t.string :name
+      t.string :email
```
Comment (gabrielgaraujo) [inline db/migrate/20220816183243_create_users.rb:5]: Let's add a DB constraint to make this unique

File: db/migrate/20220816183243_create_users.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+      t.string :password
+      t.string :first_name
```
Comment (gabrielgaraujo) [inline db/migrate/20220816183243_create_users.rb:7]: Optional, we can make these columns mandatory by setting it to `null: false`

File: db/migrate/20220816184148_create_loans.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+class CreateLoans < ActiveRecord::Migration[7.0]
+  def change
+    create_table :loans do |t|
+      t.string :loan_type
+      t.string :creditor
```
Comment (gabrielgaraujo) [inline db/migrate/20220816184148_create_loans.rb:5]: This should be a foreing key `creditor_id` pointing to the `creditors` table

File: db/migrate/20220816184148_create_loans.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+      t.string :description
+      t.string :user
```
Comment (gabrielgaraujo) [inline db/migrate/20220816184148_create_loans.rb:7]: This should be a foreing key `user_id` pointing to the `users` table

File: app/models/user.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  REGEX_PATTERN = /^[\\w!#$%&'*+/=?`{|}~^-]+(?:\\.[\\w!#$%&'*+/=?`{|}~^-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,6}$/
+  def is_email_valid? email
```
Comment (ernest-phillips) [inline app/models/user.rb:5]: @gabrielgaraujo is this what you meant by email validation?

Comment (gabrielgaraujo) [reply app/models/user.rb:5]: I meant [Active Record validations](https://guides.rubyonrails.org/active_record_validations.html)

Maybe you can use the [`format`](https://guides.rubyonrails.org/active_record_validations.html#format) helper

We can start with `URI::MailTo::EMAIL_REGEXP` and create our custom regex if we need it down the line.

File: db/migrate/20220816184148_create_loans.rb (hunk 8 above)
Comment (gabrielgaraujo) [reply db/migrate/20220816184148_create_loans.rb:7]: @ernest-phillips you can check the [`:references`](https://guides.rubyonrails.org/getting_started.html#adding-a-second-model-generating-a-model) keyword to tackle this

Comment (gabrielgaraujo) [reply db/migrate/20220816184148_create_loans.rb:7]: Both `creditor` and `user` should be a reference to their respective tables

I see you have migrations for `loans` and `users`, but no migration to create `creditors`, before you add `t.references :creditor, null: false, foreign_key: true` you should create the `Creditor` model and it's table

**Review by gabrielgaraujo — APPROVED** [review]

**Pull request merged**